web/src/providers/SupabaseAuthProvider.tsx

```diff
@@ -2,16 +2,20 @@
 
 import React, { createContext, useEffect, useState } from "react";
 import { Session } from "@supabase/supabase-js";
-
 import { supabase } from "@/services/supabase";
 import { LocalStorage } from "@/constants/storage";
+import invariant from "tiny-invariant";
 
 export const SupabaseAuthContext = createContext<{ session: Session | null } | null>(null);
 
 export default function SupabaseAuthProvider({ children }: { children: React.ReactNode }) {
+
   const [session, setSession] = useState<Session | null>(null);
 
   useEffect(() => {
+
+    invariant(supabase, "Supabase is undefined");
```

Comment:
not good sir to adapt this from UX perspective, crashes whole app when missing, better to show toast/error when it is actually interacted and missing as visitor is long before this flow

```suggestion
    invariant(supabase, "Supabase is undefined");
```